### ISSUE TITLE
Multiple copies of a file are created 

### ISSUE THREAD

**Issue by GrouchyGaijin** [skilion/onedrive#171]
This is really odd and really irritating.  I am hoping someone can explain why I am getting this error.
If I add a file, say **FileX.pd**f to the OneDrive folder on my Linux laptop it is uploaded to OneDrive as it should be.
However, moments after the upload it will be uploaded again only this time as **FIleX-john-laptop.pdf**
This will be downloaded to the OneDrive folder on my laptop where it will again be uploaded as **FileX-john-laptop-john-laptop.pdf**

The cycle will continue until OneDrive for Linux crashes and I have a ton of files, each one with another instance of john-laptop appended to the name.

What could be causing this and how can I fix it?

Thank you

**Comment by skilion** [comment]
Check that you are using the latest commit, 79ed0a7883678f094faf199f20be0661db0ee11f solved a problem that would cause that behaviour

**Comment by GrouchyGaijin** [comment]
Thank you, that seems to have solved the problem
On Tue, 2017-03-21 at 08:04 -0700, skilion wrote:
> Check that you are using the latest commit, 79ed0a7 solved a problem
> that would cause that behaviour
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
-- 


John

**Comment by GrouchyGaijin** [comment]
**Update: I downloaded the zip file and ran `make`and `make install`.**
I thought the problem was solved, but today I added a new document to my OneDrive folder and once again started having multiple copies (file-john-laptop, file-john-laptop-john-laptop, file-john-laptop-john-laptop-john-laptop) appear online and on my local machine.

**Comment by rubenqba** [comment]
Same problem here, I update last night

**Comment by skilion** [comment]
I cannot reproduce the problem

**Comment by rubenqba** [comment]
Sorry, but the same behavior is maintained. To reproduce, copy any file in OneDrive shared directory, after a few seconds it begins to replicate.

Here I attach a fragment of logs:

```
mar 24 15:00:04 mlpr1-desktop onedrive[16557]: Moving: slides_chap03.pdf -> ./slides_chap03.pdf
mar 24 15:00:04 mlpr1-desktop onedrive[16557]: ./slides_chap03.pdf: No such file or directory
mar 24 15:00:04 mlpr1-desktop onedrive[16557]: Downloading: ./slides_chap03.pdf
mar 24 15:00:05 mlpr1-desktop onedrive[16557]: Uploading: ./slides_chap03-mlpr1-desktop.pdf
```

**Comment by rubenqba** [comment]
I can confirm that problem is not solved. This is the log in my laptop. It's seen that every few seconds the file added is replicated. 

```
mar 24 17:57:35 ideapad systemd[2459]: Started OneDrive Free Client.
mar 24 17:58:51 ideapad onedrive[18891]: Uploading: ./donantes.pdf
mar 24 17:59:49 ideapad onedrive[18891]: Moving: donantes.pdf -> ./donantes.pdf
mar 24 17:59:49 ideapad onedrive[18891]: ./donantes.pdf: No such file or directory
mar 24 17:59:50 ideapad onedrive[18891]: Downloading: ./donantes.pdf
mar 24 17:59:54 ideapad onedrive[18891]: Uploading: ./donantes-ideapad.pdf
mar 24 18:00:48 ideapad onedrive[18891]: Moving: donantes-ideapad.pdf -> ./donantes-ideapad.pdf
mar 24 18:00:48 ideapad onedrive[18891]: ./donantes-ideapad.pdf: No such file or directory
mar 24 18:00:48 ideapad onedrive[18891]: Downloading: ./donantes-ideapad.pdf
mar 24 18:00:49 ideapad onedrive[18891]: Uploading: ./donantes-ideapad-ideapad.pdf
mar 24 18:01:35 ideapad onedrive[18891]: Moving: donantes-ideapad-ideapad.pdf -> ./donantes-ideapad-ideapad.pdf
mar 24 18:01:35 ideapad onedrive[18891]: ./donantes-ideapad-ideapad.pdf: No such file or directory
mar 24 18:01:36 ideapad onedrive[18891]: Downloading: ./donantes-ideapad-ideapad.pdf
mar 24 18:01:37 ideapad onedrive[18891]: Uploading: ./donantes-ideapad-ideapad-ideapad.pdf
mar 24 18:02:18 ideapad onedrive[18891]: Moving: donantes-ideapad-ideapad-ideapad.pdf -> ./donantes-ideapad-ideapad-ideapad.pdf
mar 24 18:02:18 ideapad onedrive[18891]: ./donantes-ideapad-ideapad-ideapad.pdf: No such file or directory
mar 24 18:02:19 ideapad onedrive[18891]: Downloading: ./donantes-ideapad-ideapad-ideapad.pdf
mar 24 18:02:20 ideapad onedrive[18891]: Uploading: ./donantes-ideapad-ideapad-ideapad-ideapad.pdf
```

**Comment by rubenqba** [comment]
I have to do more tests, but for the moment I have been able to deduce that the problem only happens with PDF, DOCX and XLSX files. I tried with TXT and ZIP files and it does not happen.

In fact, I did the following, I made a copy of an XLSX file but the name changed it to TXT and the latter is not replicated while the first one continues to replicate.

```bash
cd /path/to/onedrive
cp some-file.xlsx some-file.txt 
```

**Comment by Tekki** [comment]
I've notices that for exactly these file types the API sometimes gives wrong metadata, for example a wrong size. @skilion : Is it possible that a wrong size, sha1 or crc32 is the cause for these uploads?

**Comment by PterX** [comment]
I agree with @Tekki

**Comment by jomoore** [comment]
My drive seems to be spawning duplicates of freshly uploaded files when it comes to the download stage.

```
7C133D96D944AFD4!8637 20160810_114128.jpg
The local item has a different modified time 2017-Apr-09 17:48:10.9644104 remote is 2017-Apr-09 05:48:10.964Z
The local item has a different hash
The local item is unsynced, renaming
The hash is not available
Downloading: 20160810_114128.jpg
```

The file was originally uploaded by onedrive from this computer, and hasn't changed since. The detected time difference may be another case where precision is a problem. I've got no idea why the hash is different though.

**Comment by ardeaf** [comment]
Same thing here.  Had uploaded an entire directory (Work) using the CLI command '''onedrive'''.  Turned on the service for my user '''systemctl --user start onedrive''' and '''systemctl --user enable onedrive'''.  

Here's journalctl from when I started the service:

'''Apr 16 00:16:53 ardeaf onedrive[28461]: Moving: Work -> ./Work
Apr 16 00:16:53 ardeaf onedrive[28461]: ./Work: No such file or directory
Apr 16 00:16:53 ardeaf onedrive[28461]: Creating directory: ./Work
Apr 16 00:16:54 ardeaf onedrive[28461]: Deleting remote item: Work/Project Lending
Apr 16 00:16:54 ardeaf onedrive[28461]: Deleting remote item: Work/Clients
Apr 16 00:16:55 ardeaf onedrive[28461]: Deleting remote item: Work/Expense
Apr 16 00:16:56 ardeaf onedrive[28461]: Deleting remote item: Work/Payments.ods
Apr 16 00:16:56 ardeaf onedrive[28461]: Deleting remote item: Work/Signature.tiff
Apr 16 00:16:56 ardeaf onedrive[28461]: Deleting remote item: Work/Clients.ods
Apr 16 00:16:56 ardeaf onedrive[28461]: Deleting remote item: Work/Program Guidelines
Apr 16 00:16:56 ardeaf onedrive[28461]: Deleting remote item: Work/Archives
Apr 16 00:16:57 ardeaf onedrive[28461]: Deleting remote item: Work/darren notes.txt
Apr 16 00:16:57 ardeaf onedrive[28461]: Deleting remote item: Work/Lunch Orders
Apr 16 00:16:58 ardeaf onedrive[28461]: Deleting remote item: Work/Signature.png
Apr 16 00:16:58 ardeaf onedrive[28461]: Deleting remote item: Work/Forms
Apr 16 00:16:59 ardeaf onedrive[28461]: Deleting remote item: Work/Sales
Apr 16 00:16:59 ardeaf onedrive[28461]: Creating remote directory: ./Work-ardeaf
Apr 16 00:16:59 ardeaf onedrive[28461]: Creating remote directory: ./Work-ardeaf/Archives
Apr 16 00:16:59 ardeaf onedrive[28461]: Uploading: ./Work-ardeaf/Archives/Archive_Archives_2016-10-01.tar.bz2'''

From there it continued to upload everything.

**Comment by tschf** [comment]
I just built from source about one week back - d65d00c ; and I'm running the service

In the web interface, I create a new document, and call it `yy`

![image](https://user-images.githubusercontent.com/1747643/29122827-a209cb8c-7ce1-11e7-9b1f-43628cc63192.png)

This then syncs as expected:

![image](https://user-images.githubusercontent.com/1747643/29122860-beee5e3e-7ce1-11e7-8de8-83b9bb8b6bcd.png)

But after a bit, without doing any changes, `yy` starts getting duplicated:

![image](https://user-images.githubusercontent.com/1747643/29122932-0a63b4fe-7ce2-11e7-8857-97b82a5c2222.png)

```
Aug 09 09:04:20 birroth onedrive[17231]: Downloading: ./yy.docx
Aug 09 09:04:21 birroth onedrive[17231]: Uploading: ./yy-birroth.docx
Aug 09 09:05:05 birroth onedrive[17231]: Deleted directory: Attachments
Aug 09 09:05:50 birroth onedrive[17231]: Moving: yy-birroth.docx -> ./yy-birroth.docx
Aug 09 09:05:50 birroth onedrive[17231]: ./yy-birroth.docx: No such file or directory
Aug 09 09:05:50 birroth onedrive[17231]: Downloading: ./yy-birroth.docx
Aug 09 09:05:51 birroth onedrive[17231]: Uploading: ./yy-birroth-birroth.docx
Aug 09 09:06:35 birroth onedrive[17231]: Moving: yy-birroth-birroth.docx -> ./yy-birroth-birroth.docx
Aug 09 09:06:35 birroth onedrive[17231]: ./yy-birroth-birroth.docx: No such file or directory
Aug 09 09:06:36 birroth onedrive[17231]: Downloading: ./yy-birroth-birroth.docx
Aug 09 09:06:36 birroth onedrive[17231]: Uploading: ./yy-birroth-birroth-birroth.docx
```

**Comment by madchap** [comment]
Checked out the code today, I experience this behavior as well.

**Comment by Jacob-Thorne** [comment]
+1

**Comment by linuxrecon** [comment]
Having the same problem. I've found out, that this only happens, when you have a document, presentation or excel file opened in your browser while having the onedrive client up and running. If I close the file in my browser, it stops immediately to produce this dozens of files.
So basically we need a way to recognize opened excel / word / powerpoint files so that the onedrive client is able to ignore them as long as they are opened.

**Comment by agrigorev** [comment]
Hello,
I also recently experienced the same behaviour, syncing files from one Linux laptop. 
@linuxrecon: are you sure that having the file stil open in the browser causes those multiple copies? If so, what can be a workaround? Close brower before starting Linux laptop?

Regards,
Andrey

**Comment by linuxrecon** [comment]
@agrigorev: I've rechecked my theory that multiple copies are created while having the file opened in browser. It turned out that this is just one factor. When I have a document opened in browser and onedrive is started with the monitor switch (-m) it constantly produces multiple files from this specific document. However, sometimes I get multiple files even with no document / file opened in browser. A workaround for me right now is not having onedrive running with the monitor option. For e.g. I do a simple onedrive command after my work is done.

**Comment by agrigorev** [comment]
@linuxrecon Hello, do you know how to disable monitoring? Should I just replace "-" switch from the command line in the settings?

**Comment by linuxrecon** [comment]
@agrigorev I simply start onedrive whenever I need it without any switches / options (for e.g. after editing a document). This means a simple and plain "onedrive" on the command line. This means that onedrive only checks one time if there were any changes and syncs them. After this, onedrive exits.

**Comment by abraunegg** [comment]
With the latest code base and #314 - I am unable to reproduce this:

* running `onedrive --monitor --verbose`
* creating a new MS Word document on OneDrive and editing it / leaving it open in the browser
* letting the onedrive client run for a few hours

```
2018-Apr-13 13:48:48.4546238 Syncing changes from this selected path: ~/OneDrive/
2018-Apr-13 13:48:48.9494442 Applying changes of Path ID: <redacted>
2018-Apr-13 13:48:49.4788716 Adding OneDrive Root to the local database
2018-Apr-13 13:48:50.3802804 OneDrive Response: '200 - OK'
2018-Apr-13 13:48:50.3821624 Downloading asdf/Document1.docx... done.
2018-Apr-13 13:48:50.3887286 Fetching details for OneDrive Root
2018-Apr-13 13:48:50.6319015 OneDrive Root exists in the database
2018-Apr-13 13:48:50.6324567 Uploading differences of .
2018-Apr-13 13:48:50.6331539 Processing root
2018-Apr-13 13:48:50.6336989 The directory has not changed
2018-Apr-13 13:48:50.634367 Processing asdf
2018-Apr-13 13:48:50.6350231 The directory has not changed
2018-Apr-13 13:48:50.6356301 Processing 1.txt
2018-Apr-13 13:48:50.6362751 The file has not changed
2018-Apr-13 13:48:50.6367962 Processing Cygwin.zip
2018-Apr-13 13:48:50.6374103 The file has not changed
2018-Apr-13 13:48:50.6379779 Processing Document1.docx
2018-Apr-13 13:48:50.6386485 The file has not changed
2018-Apr-13 13:48:50.6392263 Processing .git
2018-Apr-13 13:48:50.6398743 The directory has not changed
2018-Apr-13 13:48:50.640482 Processing file.txt
2018-Apr-13 13:48:50.6412003 The file has not changed
2018-Apr-13 13:48:50.6417531 Processing .git2
2018-Apr-13 13:48:50.6424541 The directory has not changed
2018-Apr-13 13:48:50.6431012 Processing file.txt
2018-Apr-13 13:48:50.6437551 The file has not changed
2018-Apr-13 13:48:50.6443318 Uploading new items of .
2018-Apr-13 13:48:50.6465619 Skipping item - invalid name (Microsoft Naming Convention): ./COM7
2018-Apr-13 13:48:51.4970081 Applying changes of Path ID: <redacted>
```
```
[root@test-box OneDrive]# ls -laR
.:
total 20
drwxr-xr-x 5 root root 4096 Apr 13 11:12 .
dr-xr-x--- 6 root root 4096 Apr 13 07:26 ..
drwxr-xr-x 2 root root 4096 Apr 13 13:41 asdf
-rw-r--r-- 1 root root    0 Apr 13 11:12 COM7
drwxr-xr-x 2 root root 4096 Apr 13 11:10 .git
drwxr-xr-x 2 root root 4096 Apr 13 11:11 .git2

./asdf:
total 39272
drwxr-xr-x 2 root root     4096 Apr 13 13:41 .
drwxr-xr-x 5 root root     4096 Apr 13 11:12 ..
-rw-r--r-- 1 root root        8 Apr 13 08:20 1.txt
-rw-r--r-- 1 root root 40187613 Oct 12  2017 Cygwin.zip
-rw-r--r-- 1 root root    10058 Apr 13 13:48 Document1.docx

./.git:
total 12
drwxr-xr-x 2 root root 4096 Apr 13 11:10 .
drwxr-xr-x 5 root root 4096 Apr 13 11:12 ..
-rw-r--r-- 1 root root    5 Apr 13 11:10 file.txt

./.git2:
total 12
drwxr-xr-x 2 root root 4096 Apr 13 11:11 .
drwxr-xr-x 5 root root 4096 Apr 13 11:12 ..
-rw-r--r-- 1 root root   16 Apr 13 11:11 file.txt
[root@test-box OneDrive]# 

```


Is there any other way to reproduce this?

**Comment by abraunegg** [comment]
This is the same issue as #265, #338, #343

There appears to be an issue in the local conflict handling routine in the original codebase. This will be the next item I look at.

**Comment by abraunegg** [comment]
@GrouchyGaijin 
Please can you provide an update on this issue. If it is no longer valid, please can you close this case.

**Comment by abraunegg** [comment]
Considering this issue closed due to no response from original submitter.

If you have further issues please or come back to this, open a new case here: https://github.com/abraunegg/onedrive
